File: benchmarks/Matmul.java
public class Matmul {
    public static void main(String[] args) {
        int n = 512;
        double[] A = new double[n*n], B = new double[n*n], C = new double[n*n];
        for (int i = 0; i < n*n; i++) { A[i] = i % 7; B[i] = i % 5; }
        for (int i = 0; i < n; i++)
            for (int k = 0; k < n; k++) {
                double aik = A[i*n + k];
                int ci = i*n, bk = k*n;
                for (int j = 0; j < n; j++) C[ci + j] += aik * B[bk + j];
            }
        double s = 0; for (int i = 0; i < n; i++) s += C[i*n + i];
        System.out.println((long) s);
    }
}

File: benchmarks/matmul.rs
fn main() {
    let n = 512usize;
    let mut a = vec![0.0f64; n*n];
    let mut b = vec![0.0f64; n*n];
    let mut c = vec![0.0f64; n*n];
    for i in 0..n*n { a[i] = (i % 7) as f64; b[i] = (i % 5) as f64; }
    for i in 0..n {
        for k in 0..n {
            let aik = a[i*n + k];
            for j in 0..n { c[i*n + j] += aik * b[k*n + j]; }
        }
    }
    let mut s = 0.0f64; for i in 0..n { s += c[i*n + i]; }
    println!("{}", s as i64);
}
